docs/configuring_tor.md

```diff
@@ -116,5 +124,47 @@ circuit.
 Activating stream isolation is very straightforward, we only require the
 specification of an additional argument:
 ```
-⛰  ./lnd --tor.socks=9050 --tor.dns=nodes.lightning.directory --tor.streamisolation
+⛰  ./lnd --tor.active --tor.streamisolation
 ```
+
+## Listening for Inbound Connections
+
+In order to listen for inbound connections through Tor, an onion service must be
+created. There are two types of onion services: v2 and v3.
+
+### v2 Onion Services
+
+v2 onion services can be created automatically by `lnd` and are currently the
+default. To do so, run `lnd` with the following arguments:
+```
+⛰  ./lnd --tor.active --tor.v2
+```
+
+This will automatically create a hidden service for your node to use to listen
+for inbound connections and advertise itself to the network. The onion service's
+private key is saved to a file named `onion_private_key` in `lnd`'s base
+directory. This will allow `lnd` to recreate the same hidden service upon
+restart. If you wish to generate a new onion service, you can simply delete this
+file. The path to this private key file can also be modified with the
+`--tor.privatekeypath` argument.
+
+### v3 Onion Services
+
+v3 onion services are the latest generation of onion services and they provide a
+number of advantages over the legacy v2 onion services. To learn more about
+these benefits, see [Intro to Next Gen Onion Services](https://trac.torproject.org/projects/tor/wiki/doc/NextGenOnions).
+
+Unfortunately, at the time of writing this, v3 onion service support is still
+at an alpha level in the Tor daemon, so we're unable to automatically set them
+up within `lnd` unlike with v2 onion services. However, they can still be run
+manually! Instructions on how to do so can be found [here](https://trac.torproject.org/projects/tor/wiki/doc/NextGenOnions#Howtosetupyourownprop224service).
+
+Once the v3 is set up, `lnd` is able to use it to listen for inbound
+connections. You'll also need the onion service's hostname in order to advertise
+your node to the network. To do so, run `lnd` with the following arguments:
+```
+⛰  ./lnd --tor.active --tor.v3 --externalip=ONION_SERVICE_HOSTNAME
+```
+
+Once v3 onion service support is stable, `lnd` will be updated to also
+automatically set up v3 onion services.
```

Comment:
These docs are great! :) I might suggest adding a sample torrc for v3 nodes as well, it may not be immediately obvious for everyone wrt. what needs to change from the sample v2 torrc given above.

Reply:
Docs added :)